pkg/compose/watch.go

```diff
@@ -77,3 +91,23 @@ func (s *composeService) Watch(ctx context.Context, project *types.Project, serv
 
 	return eg.Wait()
 }
+
+func debounce(ctx context.Context, interval time.Duration, input chan string, fn func(services []string)) {
+	services := utils.Set[string]{}
+	timer := time.NewTimer(interval)
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case service := <-input:
+			timer.Reset(interval)
```

Comment:
Since we don't know if the timer has fired here, we need to stop it/drain first:
```go
if !timer.Stop() {
	<-timer.C
}
timer.Reset(interval)
```

https://pkg.go.dev/time#Timer.Reset